### DISCUSSION TITLE
How can we create holes from C#

### DISCUSSION THREAD

**Discussion by jeffercize** [TokisanGames/Terrain3D#366] [Questions] [answered]
I'm working on a procedurally generated terrain and have that part working great but I'm trying to generate a hole at certain specific points during generation. I understand the general format of the control_map from the Control Map Format page and usage of Terrain3DStorage, but believe I'm missing a critical part of how to interact with them correctly to add a hole.

Below is a small snippet of how I'm currently attempting to interact with the control map to add a hole, any advice or guidance on how to get this to work as a general rule would be appreciated:

    for (int x = 100; x < 200; x++)
    {
       for (int y = 100; y < 200; y++)
       {
          terrain.AsGodotObject().Get("storage").AsGodotObject().Call("set_pixel", 1, new Vector3(x, y, 0), (1 & 0x1) << 2);
       }
    }
    terrain.AsGodotObject().Get("storage").AsGodotObject().Call("force_update_maps", 1);

		
Thanks!

**Comment by TokisanGames** [comment, marked as answer]
Y is up. Use Z.

1&1 is 1.

Although you don't need it here, read the latest API for Terrain3DUtil and consider using a nightly build.

**Reply by jeffercize** [reply]
Got it thanks, I just assumed it was the same as when I was manipulating the image to use for the height map but it makes sense it would be like this for working in 3D.

"1&1 is 1." Haha yeah sorry I tried to simplifying my code to make it easier to pin-point the issue and I didnt think about replacing that variable making it look a little silly.

I looked at the Terrain3DUtil before but totally missed the details on bitwise-ORing and the as_uint functions those are super helpful, Thanks!
